selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java

```diff
@@ -213,22 +213,53 @@ public void updateFile(String workspaceId, String pathToFile, String content) th
     }
   }
 
-  private String getWsAgentUrl(String workspaceId) throws Exception {
-    Workspace workspace = workspaceServiceClient.getById(workspaceId);
-    workspaceServiceClient.ensureRunningStatus(workspace);
-
-    Map<String, ? extends Machine> machines =
-        workspaceServiceClient.getById(workspaceId).getRuntime().getMachines();
-    for (Machine machine : machines.values()) {
-      if (containsWsAgentServer(machine)) {
-        Server wsAgentServer = machine.getServers().get(SERVER_WS_AGENT_HTTP_REFERENCE);
-        if (wsAgentServer != null) {
-          return wsAgentServer.getUrl() + "/project";
-        } else {
-          throw new RuntimeException("Workspace agent server is null");
-        }
-      }
-    }
-    throw new RuntimeException("Cannot find dev machine on workspace");
+  public boolean checkProjectType(String wokspaceId, String projectName, String projectType)
+      throws Exception {
+    return getProjectConfig(wokspaceId, projectName).getType().equals(projectType);
+  }
+
+  public boolean checkProjectLanguage(String workspaceId, String projectName, String language)
+      throws Exception {
+
+    return getProjectConfig(workspaceId, projectName)
+        .getAttributes()
+        .get("language")
+        .contains(language);
+  }
+
+  public boolean checkProjectLanguage(
+      String workspaceId, String projectName, List<String> languages) throws Exception {
+
+    return getProjectConfig(workspaceId, projectName)
+        .getAttributes()
+        .get("language")
+        .containsAll(languages);
+  }
+
+  public List<String> getExternalLibraries(String workspaceId, String projectName)
+      throws Exception {
+    HttpJsonResponse response =
+        requestFactory
+            .fromUrl(workspaceAgentApiEndpoint.get(workspaceId) + "java/navigation/libraries")
+            .useGetMethod()
+            .addQueryParam("projectpath", "/" + projectName)
+            .request();
+
+    return DtoFactory.getInstance()
+        .createListDtoFromJson(response.asString(), ProjectConfigDto.class)
+        .stream()
+        .map(e -> e.getName())
+        .collect(Collectors.toList());
+  }
+
+  private ProjectConfigDto getProjectConfig(String workspaceId, String projectName)
```

Comment:
**ProjectConfig** is more suitable here as more general type compare to **ProjectConfigDto** class.